kubernetes/envoy.go

```diff
@@ -0,0 +1,150 @@
+package kubernetes
+
+import "github.com/mitchellh/mapstructure"
```

Comment:
I haven't used this library before, how does it work ? Does it parse an interface{} as input into a strong type struct{} ?

Reply:
Yes, it is exactly what you mentioned.
See an example:

```
type Person struct {
    Name   string
    Age    int
    Emails []string
    Extra  map[string]string
}

// This input can come from anywhere, but typically comes from
// something like decoding JSON where we're not quite sure of the
// struct initially.
input := map[string]interface{}{
    "name":   "Mitchell",
    "age":    91,
    "emails": []string{"one", "two", "three"},
    "extra": map[string]string{
        "twitter": "mitchellh",
    },
}

var result Person
err := Decode(input, &result)
if err != nil {
    panic(err)
}

fmt.Printf("%#v", result)
```
Output:
```
Person{Name:"Mitchell", Age:91, Emails:[]string{"one", "two", "three"}, Extra:map[string]string{"twitter":"mitchellh"}}
```

You can see in action in this PR in here: https://github.com/kiali/kiali/pull/3400/files#diff-f27412a2c0f12f79f5d59029c2f5947bee29ebd0673f8ecc4718b6c92afb0316R127-R133
It decodes from `map[string]interface{}` to a `RouteDump` struct.

And after discovering this small lib, the Istio object parsing and the validations code came to my head. Because with this method, we would remove all the nasty type assertion and dealing with map[string]interface{}. You just decode the `interface{}` and then you deal with the object as usual.

One good thing with this lib is that you can skip attributes you don't want. So you don't have to copy the whole structure. You create the struct with the part you only care of. That is really convenient for the envoy structs because you pick the only few field you need.

Reply:
+1
Yes, I was also exactly to update at some point that parsing we had with the interfaces in other places.